cEP-0002.md

```diff
@@ -0,0 +1,402 @@
+# Next Gen Bear Design
+
+| Metadata     |                                         |
+| ------------ |-----------------------------------------|
+| cEP          | 2                                       |
+| Version      | 0.1                                     |
+| Title        | Next gen bear design                    |
+| Authors      | Udayan Tandon <udayan12167@iiitd.ac.in> |
+| Status       | Proposed                                |
+| Type         | Process                                 |
+
+# Abstract
+
+This document describes how the next generation bears will be designed and
+implemented.
+
+This cEP describes a new implementation of bears in coala which should make
+ them simpler and make their design more intuitive and better.
+
+# Introduction
+
+Currently the bear design is clunky and bad. The current design makes global
+bears dependant on local bears through it's queuing mechanism. It uses
```

Comment:
>The current design makes global bears ~~dependant~~ **dependent** on local bears through it's queuing mechanism

and then

> global bears are done with their execution and local bears can begin

This shows that local bears are dependent on global bears and not the other way around.

Finally this contradicts the last statement.
>if local bears fail to run it stalls the global bears

So among 3, one of them seems incorrect. Correct me if I misunderstood or this was intentional? @Udayan12167